### PR TITLE
Milestone 2

### PR DESCRIPTION
**Pull request by Kubci**
Added facade for Tire
Added service for Tire
Scenario purchase tire added
Added data cache for immutable TireProperties and TireManufacturer
Added mapping service
Added config files

### CODE REVIEW

File: api/src/main/java/api/TireFacade.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.Set;
+
+/***
+ * @author Jakub Palenik, 422453@mail.muni.cz
+ */
+
+public interface TireFacade {
+
+    /**
+     * Self explanatory
```
Comment (martinzilak) [inline api/src/main/java/api/TireFacade.java]: Please write proper documentation, few words should be enough for self-explanatory functions.

File: backend/src/main/java/cz/muni/fi/pa165/pneuservis/backend/dao/TireManagerDao.java [hunk 2]
```diff
@@ -38,7 +41,8 @@
      * @return
      * @throws IllegalArgumentException on rite or tire.attribute == null
      */
-    Tire findTireByAttrs(Tire tire) throws IllegalArgumentException;
+    //Probably should be private method
```
Comment (martinzilak) [inline backend/src/main/java/cz/muni/fi/pa165/pneuservis/backend/dao/TireManagerDao.java:44]: Why?

File: services/src/main/java/config/MainConfig.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import services.MappingService;
+
+/***
+ * @author Jakub Palenik, 422453@mail.muni.cz
+ */
+
+@Configuration
+@ComponentScan(basePackageClasses = {TireFacadeImpl.class})
```
Comment (martinzilak) [inline services/src/main/java/config/MainConfig.java:14]: We should probably use a dedicated marker class, to stay consistent with [this](https://github.com/tinnyx/PA165-Pneuservis/blob/47743f6feb2724dd104ed6e024e49a9c556fd8af/backend/src/main/java/cz/muni/fi/pa165/pneuservis/backend/PersistenceApplicationContext.java#L28).

File: services/src/main/java/services/MappingService.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+/***
+ * Dozer mapping class
+ *
+ * @author Jakub Palenik, 422453@mail.muni.cz
+ */
+
+public interface MappingService {
+
+    /**
+     * Self explanatory
```
Comment (martinzilak) [inline services/src/main/java/services/MappingService.java]: Please explain at least in few words anyway.

File: services/src/main/java/services/TireService.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.Set;
+
+
+/***
+ * @author Jakub Palenik, 422453@mail.muni.cz
+ */
+public interface TireService {
+
+    /**
+     * Self explanatory
```
Comment (martinzilak) [inline services/src/main/java/services/TireService.java]: Please explain at least in few words anyway.

File: services/src/main/java/services/TireServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+    @Inject
+    private TireDataCache tdc;
+
+    @Inject
+    private TireManagerDao tireManagerDao;
+
+    @Override
+    public List<Tire> getAllTires() {
+        return tireManagerDao.retrieveAllTires();
```
Comment (martinzilak) [inline services/src/main/java/services/TireServiceImpl.java:38]: Shouldn't this be encapsulated in an unmodifiableList?

**Review by martinzilak — CHANGES_REQUESTED** [review]

File: backend/src/main/java/cz/muni/fi/pa165/pneuservis/backend/dao/TireManagerDao.java (hunk 2 above)
Comment (Kubci) [reply backend/src/main/java/cz/muni/fi/pa165/pneuservis/backend/dao/TireManagerDao.java:44]: Method is only used within the class

Comment (martinzilak) [reply backend/src/main/java/cz/muni/fi/pa165/pneuservis/backend/dao/TireManagerDao.java:44]: Okay then, that sounds correct.

File: services/src/main/java/services/TireServiceImpl.java (hunk 6 above)
Comment (Kubci) [reply services/src/main/java/services/TireServiceImpl.java:38]: true dat

**Review by martinzilak — APPROVED** [review]

**Pull request merged**